bugsnag-android-core/src/test/java/com/bugsnag/android/EventSerializationTest.kt

```diff
@@ -0,0 +1,61 @@
+package com.bugsnag.android
+
+import com.bugsnag.android.BugsnagTestUtils.generateImmutableConfig
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import org.junit.runners.Parameterized.Parameter
+import org.junit.runners.Parameterized.Parameters
+import java.util.Date
+
+@RunWith(Parameterized::class)
+internal class EventSerializationTest {
+
+    companion object {
+        @JvmStatic
+        @Parameters
+        fun testCases(): Collection<Pair<Event, String>> {
+            return generateSerializationTestCases("event",
+                createEvent(),
+
+                // custom context
+                createEvent { it.context = "ExampleContext" },
+
+                // custom grouping hash
+                createEvent { it.groupingHash = "herpderp" },
+
+                // custom severity
+                createEvent { it.severity = Severity.INFO },
+
+                // session included
+                createEvent {
+                    val user = User("123", "foo@example.com", "Joe")
+                    it.session = Session("123", Date(0), user, false)
+                },
+
+                // threads included
+                createEvent {
+                    val stacktrace = Stacktrace(arrayOf(), emptySet())
+                    it.threads = mutableListOf(Thread(5, "main", "android", true, stacktrace))
+                }
```

Comment:
It seems like several tests have been removed in the refactor: different `HandledState`s, session excluded, threads disabled.

Other fields have never been included in the test - `Error`, `App`, `Device`, `Metadata` - I understand there are separate tests for these nested objects, but shouldn't we include a test to make sure the field included in Event?

Reply:
`HandledState` is covered in the 'custom severity' test case.

I'll create an extra test case that verifies that something is serialized for breadcrumbs, app, device, user, metadata, and exceptions.